kayak_core/src/context.rs

```diff
@@ -364,20 +468,79 @@ impl KayakContext {
         }
     }
 
-    pub fn set_global_state<T: resources::Resource>(&mut self, state: T) {
-        self.global_state.insert(state);
+    /// Set a value that's accessible to all widgets
+    ///
+    /// Values should be type-unique. Setting an `i32` value, for example, allows another widget
+    /// to overwrite that value by adding their own global `i32` value, whether or not it was intentional.
+    /// If this is not desired, an easy solution is to use the [newtype](https://doc.rust-lang.org/rust-by-example/generics/new_types.html)
+    /// pattern.
+    ///
+    /// Widgets are not automatically bound to this global. You will have to bind to it manually
+    /// (as long as the value is a `Binding<T>`) using [`bind`](Self::bind).
+    ///
+    /// # Arguments
+    ///
+    /// * `value`: The value to set
+    ///
+    /// # Examples
+    ///
+    /// ```ignore
+    /// struct MyCount(i32);
+    ///
+    /// #[widget]
+    /// fn MyWidget() {
+    ///   context.set_global(MyCount(123));
+    /// }
+    /// ```
+    ///
+    /// You may also want to bind the widget to a global, so that when the global is changed,
+    /// the widget will re-render. This can be done by binding to the global.
+    ///
+    /// ```ignore
+    /// use kayak_core::bind;
+    ///
+    /// #[derive(Clone, PartialEq)] // <- Required by `bind`
+    /// struct MyCount(i32);
+    ///
+    /// #[widget]
+    /// fn MyWidget() {
+    ///   let bound_count = bind(MyCount(123));
+    ///   context.bind(&bound_count);
+    ///   context.set_global(bound_count);
+    /// }
+    /// ```
+    pub fn set_global<T: resources::Resource>(&mut self, value: T) {
+        self.global_state.insert(value);
     }
 
-    pub fn get_global_state<T: resources::Resource>(
```

Comment:
Thoughts on renaming this? Calling it `global_state` might be a bit confusing since it's only a "state" (as we define it elsewhere) when the given resource is a `Binding<T>` and a widget actually binds to it.

Reply:
Yeah I think this sounds fine. :)